drf_jsonapi/mixins.py

```diff
@@ -118,7 +118,10 @@ def process_relationships(self, relationship_data, resource):
                 data, many=handler.many
             )
 
-            handler.set_related(resource, related_resources)
+            try:
+                handler.set_related(resource, related_resources, request)
```

Comment:
I don't love this try/except here. Can we make `set_related` deal with the case where there's no `request` argument, or no?

Reply:
@chrisbrantley Late Late Reply, but...


Is this something that we can add a deprecation warning (https://docs.python.org/3/library/exceptions.html#DeprecationWarning) on, so that in the next (first?) major version, it can be removed while giving end users a heads-up? I don't mind adding a warning on a later PR, just an idea.

Since this repo is OSS, it'd be nice to follow some common standards such as this.

Reply:
@morgandenning Excellent suggestion. I just pushed a commit that adds those deprecation warnings.